docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.install

```diff
@@ -246,3 +246,106 @@ function hs_paragraph_types_update_10006() {
     }
   }
 }
+
+/**
+ * Set new config for collection color wells.
+ */
+function hs_paragraph_types_update_10007(&$sandbox) {
+  $paragraph_storage = \Drupal::entityTypeManager()->getStorage('paragraph');
+  if (empty($sandbox['ids'])) {
+
+    // Import all config before updating existing paragraphs.
+    _hs_paragraph_types_well_color_config();
+
+    // Get list of collection paragraphs.
+    $sandbox['ids'] = $paragraph_storage->getQuery()
+      ->accessCheck(FALSE)
+      ->condition('type', ['hs_collection', 'hs_priv_collection'], 'IN')
+      ->execute();
+    $sandbox['total'] = count($sandbox['ids']);
+  }
+
+  $paragraph_ids = array_splice($sandbox['ids'], 0, 10);
+
+  // Set new field default for existing hs_collection paragraphs.
+  foreach ($paragraph_storage->loadMultiple($paragraph_ids) as $paragraph) {
+    /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+    if (
+      !$paragraph->hasField('field_paragraph_style') ||
+      !$paragraph->hasField('field_bg_color') ||
+      !$paragraph->hasField('field_bg_color_width')) {
+      continue;
+    }
+
+    // Only set new field if old field had data.
+    if ($paragraph->get('field_paragraph_style')->value) {
+      $paragraph->set('field_bg_color', 'default');
+    }
+    if ($paragraph->get('field_paragraph_style')->value == 'well-full-width') {
+      $paragraph->set('field_bg_color_width', 'full_width');
+    }
+    else {
+      $paragraph->set('field_bg_color_width', 'limited_width');
+    }
+
+    $paragraph->save();
+  }
+
+  $sandbox['#finished'] = count($sandbox['ids']) ? 1 - count($sandbox['ids']) / $sandbox['total'] : 1;
+}
+
+/**
+ * Internal function to update/add well color config.
+ */
+function _hs_paragraph_types_well_color_config() {
+  $config_storage = \Drupal::service('config.storage.sync');
+  $config_factory = \Drupal::configFactory();
+
+  /** @var \Drupal\Core\Entity\EntityStorageInterface $field_config_storage */
+  $field_config_storage = \Drupal::service('entity_type.manager')->getStorage('field_config');
+
+  /** @var \Drupal\Core\Entity\EntityStorageInterface $fieldConfigStorage */
+  $field_storage = \Drupal::service('entity_type.manager')->getStorage('field_storage_config');
+
+  // If storage does not yet exist, create it.
+  $storage_to_create = [
+    'paragraph.field_bg_color' => 'field.storage.paragraph.field_bg_color',
+    'paragraph.field_bg_color_width' => 'field.storage.paragraph.field_bg_color_width',
+  ];
+
+  foreach ($storage_to_create as $key => $fields) {
+    if (empty($field_storage->load($key))) {
+      $field_storage->createFromStorageRecord($config_storage->read($fields))->save();
+    }
+  }
```

Comment:
I've found it useful to read the configuration once and build everything from there. For example:

```
// If storage does not yet exist, create it.
$bg_color_field_storage = $config_directory->read('field.storage.paragraph.field_bg_color');
if (!$field_storage->loadByName($bg_color_field_storage['entity_type'], $bg_color_field_storage['field_name'])) {
  $field_storage->create($bg_color_field_storage)->save();
}
```

Reply:
Thanks for the suggestion. I implemented it, but had to tweak it a bit:

- Used `FieldStorageConfig->loadByName()` Instead of `$field_storage->loadByName()`. `loadByName()` is a static method not available in the config object.
- used `createFromStorageRecord()` instead of `create()`, because I was getting an error with `create()`.